src/language/lexer.ts

```diff
@@ -490,11 +490,10 @@ function readDigits(lexer: Lexer, start: number, firstCode: number): number {
 
   const body = lexer.source.body;
   let position = start;
-  let code = firstCode;
 
   do {
-    code = body.charCodeAt(++position);
-  } while (isDigit(code));
+    // empty
```

Comment:
Empty loops can look suspicious in code.
I moved increate to be the loop body since it's the purpose of this code to calculate position.